### ISSUE TITLE
SSMS can't retrieve status of SQL Server and SQL Agent

### ISSUE THREAD

**Issue by Hayorti** [microsoft/mssql-docker#412]
I duplicated this dockerfile so that I could install SQL Server 2017 on a Windows Server 2019 base, and operationally everything seems fine, but I'm having a (potentially) minor problem with SSMS. I can connect to the container, browse the databases, and create and run SQL Agent jobs, but there's a blue question mark on both the SQL Server and SQL Agent icons. Reading up, it seems like this might have something to do with Windows Firewall or WMI not being accessible remotely, but I've hit a wall.

The Windows Defender Firewall service itself is off by default, and that's what most of the WMI troubleshooting actually hinges on, so I can't think of anything else to do.

Has anyone run into this before or have any thoughts as to what might be the cause/resolution?

**Comment by forkazmoden** [comment]
Hi,

Based on my test, in order to display the status of the service, the server must have WMI installed and the user must have rights to remotely access the WMI object. Windows Management Instrumentation (WMI) is the infrastructure for management data and operations on Windows-based operating systems. If you are using a LINUX container, there is no WMI in it. But I also cannot find any way to configure this permission in a Windows container. 

To return information about the SQL Server and SQL Server Agent services, you can use the [sys.dm_server_services](https://docs.microsoft.com/en-us/sql/relational-databases/system-dynamic-management-views/sys-dm-server-services-transact-sql?view=sql-server-2017) DMV. For mssql-server service in Linux, you can use `sudo systemctl status mssql-server`.  Please refer to [Troubleshoot SQL Server on Linux](https://docs.microsoft.com/en-us/sql/linux/sql-server-linux-troubleshooting-guide?view=sql-server-2017).

> select servicename,status_desc from sys.dm_server_services
```
servicename                               status_desc
SQL Server (MSSQLSERVER)	          Running
SQL Server Agent (MSSQLSERVER)	          Stopped
```

**Comment by Hayorti** [comment]
Yes, I'm aware of the WMI requirement, but that knowledge hasn't brought me any closer to a solution. Both MSSQL services are running, as is the WMI service, and the latter is definitely returning data, but because I'm using SQL authentication, it's not like I can grant the SSMS login user remote access to WMI.

**Comment by amitmsft** [comment]
We are looking into how we can make the status show up in SSMS and other tools that we have which depend on components that depend on Windows components. For now, we recommend that you use systemctl commands to query the service status.

**Comment by Hayorti** [comment]
Thanks for the info, @amitmsft. To clarify, though, I'm running it in a Windows Server 2019 container, not a Linux one. Any idea when you might have a solution, or where it would be posted once you do?

**Comment by Chesky7** [comment]
Having the same issue on a clustered SQL server, one node doesn't have the blue question mark and the other does, both servers have wmi services on and firewall off.
